ouroboros-network/src/Ouroboros/Network/NodeToNode.hs

```diff
@@ -538,3 +548,74 @@ localNetworkErrorPolicy = ErrorPolicies {
   where
     ourBug :: SuspendDecision DiffTime
     ourBug = Throw
+
+-- | Simple responder control function that discards the Protocol result.
+simpleResponderControl :: forall m a. (MonadSTM m)
+                       => (NodeToNodeProtocols -> MiniProtocolResponderControl m a)
+                       -> m ()
+simpleResponderControl rspFn = forever $
+    -- Fetch, but ignore the result of any miniprotocol so that it can
+    -- be restarted incase the initiator decides to talk to us again.
+    void $ atomically $ fetchResponderResult ChainSyncWithHeadersPtcl
+               `orElse` fetchResponderResult BlockFetchPtcl
+               `orElse` fetchResponderResult TxSubmissionPtcl
+  where
+    fetchResponderResult :: MonadSTM m => NodeToNodeProtocols -> STM m a
+    fetchResponderResult ptcl =
+        let (MiniProtocolResponderControl action) = rspFn ptcl in
```

Comment:
I'd prefer `case rspFn ptcl of MiniProtocolResponderControl action -> action` - though it's just matter of taste.

It does not directly apply to Haskell, but in the core language: `case` expressions drive the evaluation and `let`s are where memory allocations happen.

Reply:
`let` isn't used any more after the addition of `getMiniProtocolResponderControl`.